### ISSUE TITLE
SSL models: Mocov2 is actually Mocov1 by default.

### ISSUE THREAD

**Issue by MkuuWaUjinga** [PyTorchLightning/lightning-bolts#612]
## 🐛 Bug
Reading through the docs and the naming of the class `Mocov2` suggests that the default is per https://arxiv.org/abs/2003.04297. However, the usage of an mlp projection head and the reduced temperature parameters need to be set explicitly. Also, the cosine learning rate scheduler is missing.

### Expected behavior
I'd expect the default initialization of the `Mocov2` class to be in-line with the proposed config in the paper. While the temperature and the mlp projection head can be configured, the learning rate scheduler is simply missing and needs to be added in `configure_optimizers`.
Happy to make a PR.
